poetry/masonry/builders/builder.py

```diff
@@ -112,10 +102,18 @@ def find_excluded_files(self):  # type: () -> Set[str]
         return result
 
     def is_excluded(self, filepath):  # type: (Union[str, Path]) -> bool
-        if not isinstance(filepath, basestring):
-            filepath = filepath.as_posix()
+        exclude_path = Path(filepath)
+
+        while True:
```

Comment:
Why not `while len(exclude_path.parts) > 1`?

I try to avoid `while True` statements to have a more explicit condition.

Reply:
Because - at least in theory - the root folder could be excluded as well. (Which doesn't make sense, if you want  to build a package, but people have sometimes strange ideas).

Of course I could check this if I leave the loop. If you would like this more, I could implement it.

Reply:
You know what? We can leave it like this for now. I don't think this is too much of an issue.